docs/source/tutorials/Microlensing.ipynb

```diff
@@ -0,0 +1,318 @@
+    "        brightness = self.src.brightness(beta_x, beta_y, params)\n",
+    "        # Compute the baseline (unlensed) brightness of the source\n",
+    "        baseline_brightness = self.src.brightness(theta_x, theta_y, params)\n",
+    "        # Return the lensed image [n_pix x n_pix], and magnification\n",
+    "        return brightness, brightness.mean() / baseline_brightness.mean()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can visualize the structure of our simulator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim = Microlens(point_lens, src, theta_x=theta_x, theta_y=theta_y, z_s=0.0)\n",
+    "sim.graph(True, True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can set up the parameters for our simulation. We can look at the order of the parameters expected by our simulator by looking at `x_order`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim.x_order"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Microlensing Physical Parameters\n",
+    "th_ein = 0.3  # Einstein radius in arcsec\n",
+    "\n",
+    "# Microlensing Model Parameters\n",
+    "t0 = 0.2  # time at peak magnification\n",
+    "u0 = 0.5  # minimum impact parameter (u(t=t_0)) in units of th_ein\n",
+    "tE = 0.05  # Einstein timescale\n",
+    "# rho = 3.0  # source size in units of lens Einstein radii\n",
+    "rho = 2.0  # source size in units of lens Einstein radii\n",
+    "\n",
+    "gamma = 0.0  # linear limb darkening coefficient"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "B = 64  # Batch size\n",
+    "ts = torch.linspace(-6 * tE + t0, 6 * tE + t0, B).view(-1, 1)  # Create time grid\n",
+    "\n",
+    "# Calculate source position at each time in arcsec\n",
+    "x0s = (ts - t0) / (tE) * th_ein  # Shape is [B, 1]\n",
+    "\n",
+    "# Since source is assumed radially symmetric, can set all motion in the x direction\n",
+    "y0s = -u0 * th_ein * torch.ones_like(x0s)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = torch.tensor(\n",
+    "    [\n",
+    "        0.0,  # z_l\n",
+    "        0.0,  # x0\n",
+    "        0.0,  # y0\n",
+    "        th_ein,  # th_ein\n",
+    "        0.0,  # x0\n",
+    "        0.0,  # y0\n",
+    "        th_ein * rho,  # theta_s\n",
+    "        5.0,  # Ie,\n",
+    "        gamma,  # gamma\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "# Create the input tensor for the simulator. In this simple example, everything is static except for the lens position\n",
```

Comment:
If everything except the lens position `x0` is static then you can set the parameters as fixed using calls like `sim.lens.th_ein = th_ein` and then the `x` you input to the simulator only needs to include the `x0` value